src/benchmarklib/column_generator.cpp

```diff
@@ -0,0 +1,260 @@
+
+#include "storage/table.hpp"
+#include "storage/table_column_definition.hpp"
+
+namespace opossum {
+
+ColumnGenerator::ColumnGenerator() : _probability_dist(0.0, 1.0) {
+  std::random_device rd;
+  _pseudorandom_engine.seed(rd());
+}
+
+/**
+ * Generates a value vector of type int according to the given column_data distribution. The values are selected
+ * randomly.
+ * @tparam VectorType Type of vector. Could be std::vector<int> or tbb::concurrent_vector<int>
+ * @param column_data_distribution Distribution of values.
+ * @param row_count Number of rows to calculate
+ * @param allow_value Function which checks, if the random value may be added to the returned vector.
+ * @return A vector with row_count randomly selected integers for which allow_value returned true.
+ */
+// Todo: Refactor TableGenerator::generate_table to uses this function
+template <typename VectorType>
+VectorType ColumnGenerator::generate_value_vector(const ColumnDataDistribution& column_data_distribution,
+                                                  size_t row_count, const std::function<bool(int)>& allow_value) {
+  VectorType result(row_count);
+
+  auto generate_value_by_distribution_type = std::function<int(void)>{};
+
+  // generate distribution from column configuration
+  switch (column_data_distribution.distribution_type) {
+    case DataDistributionType::Uniform: {
+      auto uniform_dist = boost::math::uniform_distribution<double>{column_data_distribution.min_value,
+                                                                    column_data_distribution.max_value};
+      generate_value_by_distribution_type = [&]() {
+        const auto probability = _probability_dist(_pseudorandom_engine);
+        return static_cast<int>(std::floor(boost::math::quantile(uniform_dist, probability)));
+      };
+      break;
+    }
+    case DataDistributionType::NormalSkewed: {
+      auto skew_dist = boost::math::skew_normal_distribution<double>{column_data_distribution.skew_location,
+                                                                     column_data_distribution.skew_scale,
+                                                                     column_data_distribution.skew_shape};
+      generate_value_by_distribution_type = [&]() {
+        const auto probability = _probability_dist(_pseudorandom_engine);
+        return static_cast<int>(std::round(boost::math::quantile(skew_dist, probability) * 10));
+      };
+      break;
+    }
+    case DataDistributionType::Pareto: {
+      auto pareto_dist = boost::math::pareto_distribution<double>{column_data_distribution.pareto_scale,
+                                                                  column_data_distribution.pareto_shape};
+      generate_value_by_distribution_type = [&]() {
+        const auto probability = _probability_dist(_pseudorandom_engine);
+        return static_cast<int>(std::floor(boost::math::quantile(pareto_dist, probability)));
+      };
+      break;
+    }
```

Comment:
You could add a 

```cpp
default {
  Fail("Unsupported data distribution type.");
}
```

Reply:
The column_generator is probably also an artefact...